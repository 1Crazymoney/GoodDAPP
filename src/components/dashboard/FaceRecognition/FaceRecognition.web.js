// @flow
import React, { createRef } from 'react'
import loadjs from 'loadjs'
import { StyleSheet, View } from 'react-native'
import API from '../../../lib/API/api'
import GDStore from '../../../lib/undux/GDStore'
import { normalize } from 'react-native-elements'
import logger from '../../../lib/logger/pino-logger'
import { Camera, getResponsiveVideoDimensions } from './Camera.web'
import Config from '../../../config/config'
import { Wrapper, CustomButton, Section } from '../../common'
import { initializeAndPreload, capture, type ZoomCaptureResult } from './Zoom'
import type { DashboardProps } from '../Dashboard'

const log = logger.child({ from: 'FaceRecognition' })

type FaceRecognitionProps = DashboardProps & {
  screenProps: any,
  store: Store
}

type State = {
  showZoomCapture: boolean,
  ready: boolean
}

class FaceRecognition extends React.Component<FaceRecognitionProps, State> {
  state = {
    showZoomCapture: false,
    ready: false
  }

  containerRef = createRef()
  width = 720
  height = 0

  async componentDidMount() {
    try {
      await this.loadZoomSDK()
      // eslint-disable-next-line no-undef
      let loadedZoom = ZoomSDK
      log.info('ZoomSDK loaded', loadedZoom)
      await initializeAndPreload(loadedZoom) // TODO: what  to do in case of init errors?
      log.info('ZoomSDK initialized and preloaded', loadedZoom)
      this.setState({ ready: true })
    } catch (e) {
      log.error(e)
    }
    this.setWidth()
  }

  loadZoomSDK = async (): Promise<void> => {
    global.exports = {} // required by zoomSDK
    const server = Config.serverUrl
    log.info({ server })
    const zoomSDKPath = '/ZoomAuthentication.js/ZoomAuthentication.js'
    log.info(`loading ZoomSDK from ${zoomSDKPath}`)
    return loadjs(zoomSDKPath, { returnPromise: true })
  }

  onCameraLoad = async (track: MediaStreamTrack) => {
    let captureOutcome: ZoomCaptureResult

    try {
      captureOutcome = await capture(track) // TODO: handle capture errors.
    } catch (e) {
      log.error(`Failed on capture, error: ${e}`)
    }
    // this.props.screenProps.onCaptureComplete(captureOutcome)
    log.info({ captureOutcome })
    await this.enroll(captureOutcome)
  }

  setWidth = () => {
    const containerWidth =
      (this.containerRef && this.containerRef.current && this.containerRef.current.offsetWidth) || this.width
    this.width = Math.min(this.width, containerWidth)
    this.height = window.innerHeight > window.innerWidth ? this.width * 1.77777778 : this.width * 0.5625

    this.width = 720
    this.height = 1280
  }

  enroll = async (captureResult: ZoomCaptureResult) => {
    log.info({ captureResult })
    try {
      await API.enroll(captureResult)
      this.props.store.set('currentScreen')({
        dialogData: {
          visible: true,
          title: 'Success',
          message: `You've successfully passed face recognition`,
          dismissText: 'YAY!',
          onDismiss: this.props.screenProps.goToRoot
        },
        loading: true
      })
    } catch (e) {
      log.warn('FaceRecognition failed', e)
    }
  }

  render() {
    const { store }: FaceRecognitionProps = this.props
    const { fullName } = store.get('profile')
    const showZoomCapture = this.state.showZoomCapture
    return (
      <Wrapper>
        {!showZoomCapture && (
          <View style={styles.topContainer}>
            <Section.Title>{`${fullName},\n Just one last thing...`}</Section.Title>
            <Section.Text style={styles.description}>
              {"In order to give you a basic income we need to make sure it's really you"}
            </Section.Text>
          </View>
        )}
        {!showZoomCapture && (
          <View style={styles.bottomContainer}>
            <CustomButton
              mode="contained"
              onPress={this.setState({ showZoomCapture: true })}
              loading={this.props.store.get('currentScreen').loading}
            >
              Quick Face Recognition
            </CustomButton>
          </View>
        )}
        {showZoomCapture && (
          <View>
            <Section style={styles.bottomSection}>
              <div id="zoom-parent-container" style={getVideoContainerStyles()}>
                <div id="zoom-interface-container" style={{ position: 'absolute' }} />
                {this.state.ready && <Camera height={this.height} onLoad={this.onCameraLoad} />}
              </div>
            </Section>
          </View>
        )}
      </Wrapper>
    )
  }
}

const styles = StyleSheet.create({
  topContainer: {
    display: 'flex',
    flex: 1,
    justifyContent: 'space-evenly',
    paddingTop: normalize(30)
  },
  bottomContainer: {
    display: 'flex',
    flex: 1,
    paddingTop: normalize(20),
    justifyContent: 'flex-end'
  },
  description: {
    fontSize: normalize(20)
  },
  bottomSection: {
    flex: 1,
    backgroundColor: '#fff'
  }
})

const getVideoContainerStyles = () => ({
  ...getResponsiveVideoDimensions(),
  marginLeft: 'auto',
  marginRight: 'auto',
  marginTop: 0,
  marginBottom: 0
})

export default GDStore.withStore(FaceRecognition)
