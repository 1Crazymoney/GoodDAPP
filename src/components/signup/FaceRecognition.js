// @flow
import React from 'react'
import { StyleSheet } from 'react-native'
import { Wrapper, Title, Description } from './components'
import { normalize } from 'react-native-elements'

type Props = {
  screenProps: any
}
type State = {}
export default class FaceRecognition extends React.Component<Props, State> {
  handleSubmit = () => {
    this.props.screenProps.doneCallback({ isEmailConfirmed: true })
  }
  render() {
    return (
      <Wrapper valid={true} handleSubmit={this.handleSubmit} submitText="Quick face recognition">
        <Title>{`${this.props.screenProps.data.fullName},\n Just one last thing...`}</Title>
        <Description style={styles.description}>
          {"We want to give you a basic income every day, so we need to make sure it's you"}
        </Description>
      </Wrapper>
    )
  }
}

const styles = StyleSheet.create({
  description: {
    fontSize: normalize(20)
  }
})
