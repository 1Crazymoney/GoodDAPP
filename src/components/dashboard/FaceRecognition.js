// @flow
import React from 'react'
import goodWallet from '../../lib/wallet/GoodWallet'
import { StyleSheet, Text, View } from 'react-native'
import { Title, Description } from '../signup/components'
import { normalize } from 'react-native-elements'
import logger from '../../lib/logger/pino-logger'
import { Button, Modal, Portal, Dialog, Paragraph } from 'react-native-paper'
import { Wrapper, NextButton } from '../common'

const log = logger.child({ from: 'FaceRecognition' })

type Props = {
  screenProps: any
}
type State = {
  dialogVisible: boolean,
  dialogData: any
}

class FaceRecognition extends React.Component<Props, State> {
  state = {
    dialogVisible: false,
    dialogData: {}
  }

  handleSubmit = () => {
    this.props.screenProps.doneCallback({ isEmailConfirmed: true })
  }

  handleClaim = async () => {
    try {
      await goodWallet.claim()
      this.setState({
        dialogVisible: true,
        dialogData: { title: 'Success', message: `You've claimed your GD`, dismissText: 'YAY!' }
      })
    } catch (e) {
      log.warn('estoooo0', e)
      this.setState({ dialogVisible: true, dialogData: { title: 'Error', message: e.message } })
    }
  }

  _handleDismissDialog = () => {
    const { title } = this.state.dialogData
    this.setState({
      dialogData: {},
      dialogVisible: false
    })
    // TODO: Improve this flow
    if (title === 'Success') this.props.screenProps.pop()
  }

  render() {
    this.props.screenProps.data = { name: 'John' }
    const { dialogVisible, dialogData } = this.state
    return (
      <Wrapper>
        <View style={styles.topContainer}>
          <Title>{`${this.props.screenProps.data.name},\n Just one last thing...`}</Title>
          <Description style={styles.description}>
            {"In order to give you a basic income we need to make sure it's really you"}
          </Description>
        </View>
        <View style={styles.bottomContainer}>
          <NextButton onPress={this.handleClaim}>Quick Face Recognition</NextButton>
        </View>

        <Portal>
          <Dialog visible={dialogVisible} onDismiss={this._handleDismissDialog} dismissable={true}>
            <Dialog.Title>{dialogData.title}</Dialog.Title>
            <Dialog.Content>
              <Paragraph>{dialogData.message}</Paragraph>
            </Dialog.Content>
            <Dialog.Actions>
              <Button onPress={this._handleDismissDialog}>{dialogData.dismissText || 'Done'}</Button>
            </Dialog.Actions>
          </Dialog>
        </Portal>
      </Wrapper>
    )
  }
}

const styles = StyleSheet.create({
  topContainer: {
    display: 'flex',
    flex: 1,
    justifyContent: 'space-evenly',
    paddingTop: normalize(30)
  },
  bottomContainer: {
    display: 'flex',
    flex: 1,
    paddingTop: normalize(20),
    justifyContent: 'flex-end'
  }
})

export default FaceRecognition
