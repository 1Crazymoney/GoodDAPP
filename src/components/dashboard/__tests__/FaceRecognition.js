import React from 'react'
import FaceRecognition from '../FaceRecognition'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

describe('FaceRecognition', () => {
  it('renders without errors', () => {
    const tree = renderer.create(<FaceRecognition screenProps={{}} />)
    expect(tree.toJSON()).toBeTruthy()
  })

  it('matches snapshot', () => {
    const component = renderer.create(<FaceRecognition screenProps={{}} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
