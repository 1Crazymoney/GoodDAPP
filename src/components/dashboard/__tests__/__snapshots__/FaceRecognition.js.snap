// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FaceRecognition matches snapshot 1`] = `
<div
  style={
    Object {
      "WebkitAlignItems": "stretch",
      "WebkitBoxAlign": "stretch",
      "WebkitBoxDirection": "normal",
      "WebkitBoxFlex": 1,
      "WebkitBoxOrient": "vertical",
      "WebkitFlexBasis": "0%",
      "WebkitFlexDirection": "column",
      "WebkitFlexGrow": 1,
      "WebkitFlexShrink": 1,
      "alignItems": "stretch",
      "borderBottomStyle": "solid",
      "borderBottomWidth": "0px",
      "borderLeftStyle": "solid",
      "borderLeftWidth": "0px",
      "borderRightStyle": "solid",
      "borderRightWidth": "0px",
      "borderTopStyle": "solid",
      "borderTopWidth": "0px",
      "boxSizing": "border-box",
      "display": "flex",
      "flexBasis": "0%",
      "flexDirection": "column",
      "flexGrow": 1,
      "flexShrink": 1,
      "marginBottom": "0px",
      "marginLeft": "0px",
      "marginRight": "0px",
      "marginTop": "0px",
      "minHeight": "0px",
      "minWidth": "0px",
      "msFlexAlign": "stretch",
      "msFlexDirection": "column",
      "msFlexNegative": 1,
      "msFlexPositive": 1,
      "msFlexPreferredSize": "0%",
      "paddingBottom": "0px",
      "paddingLeft": "0px",
      "paddingRight": "0px",
      "paddingTop": "0px",
      "position": "relative",
      "zIndex": 0,
    }
  }
>
  <div
    style={
      Object {
        "WebkitAlignItems": "stretch",
        "WebkitBoxAlign": "stretch",
        "WebkitBoxDirection": "normal",
        "WebkitBoxFlex": 1,
        "WebkitBoxOrient": "vertical",
        "WebkitFlexBasis": "0%",
        "WebkitFlexDirection": "column",
        "WebkitFlexGrow": 1,
        "WebkitFlexShrink": 1,
        "alignItems": "stretch",
        "borderBottomStyle": "solid",
        "borderBottomWidth": "0px",
        "borderLeftStyle": "solid",
        "borderLeftWidth": "0px",
        "borderRightStyle": "solid",
        "borderRightWidth": "0px",
        "borderTopStyle": "solid",
        "borderTopWidth": "0px",
        "boxSizing": "border-box",
        "display": "flex",
        "flexBasis": "0%",
        "flexDirection": "column",
        "flexGrow": 1,
        "flexShrink": 1,
        "marginBottom": "0px",
        "marginLeft": "0px",
        "marginRight": "0px",
        "marginTop": "0px",
        "minHeight": "0px",
        "minWidth": "0px",
        "msFlexAlign": "stretch",
        "msFlexDirection": "column",
        "msFlexNegative": 1,
        "msFlexPositive": 1,
        "msFlexPreferredSize": "0%",
        "paddingBottom": "0px",
        "paddingLeft": "0px",
        "paddingRight": "0px",
        "paddingTop": "0px",
        "position": "relative",
        "zIndex": 0,
      }
    }
  >
    <div
      style={
        Object {
          "WebkitAlignItems": "stretch",
          "WebkitBoxAlign": "stretch",
          "WebkitBoxDirection": "normal",
          "WebkitBoxFlex": 1,
          "WebkitBoxOrient": "horizontal",
          "WebkitBoxPack": "center",
          "WebkitFlexBasis": "0%",
          "WebkitFlexDirection": "row",
          "WebkitFlexGrow": 1,
          "WebkitFlexShrink": 1,
          "WebkitJustifyContent": "center",
          "alignItems": "stretch",
          "borderBottomStyle": "solid",
          "borderBottomWidth": "0px",
          "borderLeftStyle": "solid",
          "borderLeftWidth": "0px",
          "borderRightStyle": "solid",
          "borderRightWidth": "0px",
          "borderTopStyle": "solid",
          "borderTopWidth": "0px",
          "boxSizing": "border-box",
          "display": "flex",
          "flexBasis": "0%",
          "flexDirection": "row",
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
          "marginBottom": "0px",
          "marginLeft": "0px",
          "marginRight": "0px",
          "marginTop": "0px",
          "minHeight": "0px",
          "minWidth": "0px",
          "msFlexAlign": "stretch",
          "msFlexDirection": "row",
          "msFlexNegative": 1,
          "msFlexPack": "center",
          "msFlexPositive": 1,
          "msFlexPreferredSize": "0%",
          "paddingBottom": "0px",
          "paddingLeft": "0px",
          "paddingRight": "0px",
          "paddingTop": "0px",
          "position": "relative",
          "zIndex": 0,
        }
      }
    >
      <div
        style={
          Object {
            "WebkitAlignItems": "stretch",
            "WebkitBoxAlign": "stretch",
            "WebkitBoxDirection": "normal",
            "WebkitBoxFlex": 1,
            "WebkitBoxOrient": "vertical",
            "WebkitBoxPack": "justify",
            "WebkitFlexBasis": "0%",
            "WebkitFlexDirection": "column",
            "WebkitFlexGrow": 1,
            "WebkitFlexShrink": 1,
            "WebkitJustifyContent": "space-between",
            "alignItems": "stretch",
            "borderBottomStyle": "solid",
            "borderBottomWidth": "0px",
            "borderLeftStyle": "solid",
            "borderLeftWidth": "0px",
            "borderRightStyle": "solid",
            "borderRightWidth": "0px",
            "borderTopStyle": "solid",
            "borderTopWidth": "0px",
            "boxSizing": "border-box",
            "display": "flex",
            "flexBasis": "0%",
            "flexDirection": "column",
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "space-between",
            "marginBottom": "0px",
            "marginLeft": "0px",
            "marginRight": "0px",
            "marginTop": "0px",
            "minHeight": "0px",
            "minWidth": "0px",
            "msFlexAlign": "stretch",
            "msFlexDirection": "column",
            "msFlexNegative": 1,
            "msFlexPack": "justify",
            "msFlexPositive": 1,
            "msFlexPreferredSize": "0%",
            "paddingBottom": "1rem",
            "paddingLeft": "1rem",
            "paddingRight": "1rem",
            "paddingTop": "1rem",
            "position": "relative",
            "width": "100%",
            "zIndex": 0,
          }
        }
      >
        <div
          style={
            Object {
              "WebkitAlignItems": "stretch",
              "WebkitBoxAlign": "stretch",
              "WebkitBoxDirection": "normal",
              "WebkitBoxFlex": 1,
              "WebkitBoxOrient": "vertical",
              "WebkitBoxPack": "space-evenly",
              "WebkitFlexBasis": "0%",
              "WebkitFlexDirection": "column",
              "WebkitFlexGrow": 1,
              "WebkitFlexShrink": 1,
              "WebkitJustifyContent": "space-evenly",
              "alignItems": "stretch",
              "borderBottomStyle": "solid",
              "borderBottomWidth": "0px",
              "borderLeftStyle": "solid",
              "borderLeftWidth": "0px",
              "borderRightStyle": "solid",
              "borderRightWidth": "0px",
              "borderTopStyle": "solid",
              "borderTopWidth": "0px",
              "boxSizing": "border-box",
              "display": "flex",
              "flexBasis": "0%",
              "flexDirection": "column",
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "space-evenly",
              "marginBottom": "0px",
              "marginLeft": "0px",
              "marginRight": "0px",
              "marginTop": "0px",
              "minHeight": "0px",
              "minWidth": "0px",
              "msFlexAlign": "stretch",
              "msFlexDirection": "column",
              "msFlexNegative": 1,
              "msFlexPack": "space-evenly",
              "msFlexPositive": 1,
              "msFlexPreferredSize": "0%",
              "paddingBottom": "0px",
              "paddingLeft": "0px",
              "paddingRight": "0px",
              "paddingTop": "30px",
              "position": "relative",
              "zIndex": 0,
            }
          }
        >
          <div
            dir="auto"
            style={
              Object {
                "borderBottomWidth": "0px",
                "borderLeftWidth": "0px",
                "borderRightWidth": "0px",
                "borderTopWidth": "0px",
                "boxSizing": "border-box",
                "color": "rgba(85,85,85,1.00)",
                "direction": "ltr",
                "display": "inline",
                "fontFamily": "Helvetica, \\"sans-serif\\"",
                "fontSize": "28px",
                "fontStyle": "inherit",
                "fontVariant": "inherit",
                "fontWeight": "inherit",
                "lineHeight": "inherit",
                "marginBottom": "30px",
                "marginLeft": "0px",
                "marginRight": "0px",
                "marginTop": "0px",
                "paddingBottom": "0px",
                "paddingLeft": "0px",
                "paddingRight": "0px",
                "paddingTop": "0px",
                "textAlign": "center",
                "textDecoration": "none",
                "whiteSpace": "pre-wrap",
                "wordWrap": "break-word",
              }
            }
          >
            John,
 Just one last thing...
          </div>
          <div
            dir="auto"
            style={
              Object {
                "borderBottomWidth": "0px",
                "borderLeftWidth": "0px",
                "borderRightWidth": "0px",
                "borderTopWidth": "0px",
                "boxSizing": "border-box",
                "color": "rgba(85,85,85,1.00)",
                "direction": "ltr",
                "display": "inline",
                "fontFamily": "Helvetica, \\"sans-serif\\"",
                "fontSize": "18px",
                "fontStyle": "inherit",
                "fontVariant": "inherit",
                "fontWeight": "inherit",
                "lineHeight": "inherit",
                "marginBottom": "0px",
                "marginLeft": "0px",
                "marginRight": "0px",
                "marginTop": "30px",
                "paddingBottom": "0px",
                "paddingLeft": "0px",
                "paddingRight": "0px",
                "paddingTop": "0px",
                "textAlign": "center",
                "textDecoration": "none",
                "whiteSpace": "pre-wrap",
                "wordWrap": "break-word",
              }
            }
          >
            In order to give you a basic income we need to make sure it's really you
          </div>
        </div>
        <div
          style={
            Object {
              "WebkitAlignItems": "stretch",
              "WebkitBoxAlign": "stretch",
              "WebkitBoxDirection": "normal",
              "WebkitBoxFlex": 1,
              "WebkitBoxOrient": "vertical",
              "WebkitBoxPack": "end",
              "WebkitFlexBasis": "0%",
              "WebkitFlexDirection": "column",
              "WebkitFlexGrow": 1,
              "WebkitFlexShrink": 1,
              "WebkitJustifyContent": "flex-end",
              "alignItems": "stretch",
              "borderBottomStyle": "solid",
              "borderBottomWidth": "0px",
              "borderLeftStyle": "solid",
              "borderLeftWidth": "0px",
              "borderRightStyle": "solid",
              "borderRightWidth": "0px",
              "borderTopStyle": "solid",
              "borderTopWidth": "0px",
              "boxSizing": "border-box",
              "display": "flex",
              "flexBasis": "0%",
              "flexDirection": "column",
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "flex-end",
              "marginBottom": "0px",
              "marginLeft": "0px",
              "marginRight": "0px",
              "marginTop": "0px",
              "minHeight": "0px",
              "minWidth": "0px",
              "msFlexAlign": "stretch",
              "msFlexDirection": "column",
              "msFlexNegative": 1,
              "msFlexPack": "end",
              "msFlexPositive": 1,
              "msFlexPreferredSize": "0%",
              "paddingBottom": "0px",
              "paddingLeft": "0px",
              "paddingRight": "0px",
              "paddingTop": "20px",
              "position": "relative",
              "zIndex": 0,
            }
          }
        >
          <div
            style={
              Object {
                "WebkitAlignItems": "stretch",
                "WebkitBoxAlign": "stretch",
                "WebkitBoxDirection": "normal",
                "WebkitBoxOrient": "vertical",
                "WebkitBoxPack": "center",
                "WebkitFlexBasis": "auto",
                "WebkitFlexDirection": "column",
                "WebkitFlexShrink": 0,
                "WebkitJustifyContent": "center",
                "alignItems": "stretch",
                "backgroundColor": "rgba(85,85,85,1.00)",
                "borderBottomColor": "rgba(0,0,0,0.00)",
                "borderBottomLeftRadius": "4px",
                "borderBottomRightRadius": "4px",
                "borderBottomStyle": "solid",
                "borderBottomWidth": "0px",
                "borderLeftColor": "rgba(0,0,0,0.00)",
                "borderLeftStyle": "solid",
                "borderLeftWidth": "0px",
                "borderRightColor": "rgba(0,0,0,0.00)",
                "borderRightStyle": "solid",
                "borderRightWidth": "0px",
                "borderTopColor": "rgba(0,0,0,0.00)",
                "borderTopLeftRadius": "4px",
                "borderTopRightRadius": "4px",
                "borderTopStyle": "solid",
                "borderTopWidth": "0px",
                "boxShadow": "0px 0.75px 1.5px rgba(0,0,0,0.24)",
                "boxSizing": "border-box",
                "display": "flex",
                "flexBasis": "auto",
                "flexDirection": "column",
                "flexShrink": 0,
                "justifyContent": "center",
                "marginBottom": "0px",
                "marginLeft": "0px",
                "marginRight": "0px",
                "marginTop": "0px",
                "minHeight": "0px",
                "minWidth": "64px",
                "msFlexAlign": "stretch",
                "msFlexDirection": "column",
                "msFlexNegative": 0,
                "msFlexPack": "center",
                "msFlexPreferredSize": "auto",
                "paddingBottom": "0px",
                "paddingLeft": "0px",
                "paddingRight": "0px",
                "paddingTop": "0px",
                "position": "relative",
                "zIndex": 0,
              }
            }
          >
            <div
              data-focusable={true}
              disabled={false}
              onKeyDown={[Function]}
              onKeyPress={[Function]}
              onKeyUp={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              role="button"
              style={
                Object {
                  "WebkitAlignItems": "stretch",
                  "WebkitBoxAlign": "stretch",
                  "WebkitBoxDirection": "normal",
                  "WebkitBoxOrient": "vertical",
                  "WebkitFlexBasis": "auto",
                  "WebkitFlexDirection": "column",
                  "WebkitFlexShrink": 0,
                  "alignItems": "stretch",
                  "borderBottomLeftRadius": "4px",
                  "borderBottomRightRadius": "4px",
                  "borderBottomStyle": "solid",
                  "borderBottomWidth": "0px",
                  "borderLeftStyle": "solid",
                  "borderLeftWidth": "0px",
                  "borderRightStyle": "solid",
                  "borderRightWidth": "0px",
                  "borderTopLeftRadius": "4px",
                  "borderTopRightRadius": "4px",
                  "borderTopStyle": "solid",
                  "borderTopWidth": "0px",
                  "boxSizing": "border-box",
                  "cursor": "pointer",
                  "display": "flex",
                  "flexBasis": "auto",
                  "flexDirection": "column",
                  "flexShrink": 0,
                  "marginBottom": "0px",
                  "marginLeft": "0px",
                  "marginRight": "0px",
                  "marginTop": "0px",
                  "minHeight": "0px",
                  "minWidth": "0px",
                  "msFlexAlign": "stretch",
                  "msFlexDirection": "column",
                  "msFlexNegative": 0,
                  "msFlexPreferredSize": "auto",
                  "msTouchAction": "manipulation",
                  "overflowX": "hidden",
                  "overflowY": "hidden",
                  "paddingBottom": "0px",
                  "paddingLeft": "0px",
                  "paddingRight": "0px",
                  "paddingTop": "0px",
                  "position": "relative",
                  "touchAction": "manipulation",
                  "zIndex": 0,
                }
              }
              tabIndex="0"
            >
              <div
                style={
                  Object {
                    "WebkitAlignItems": "center",
                    "WebkitBoxAlign": "center",
                    "WebkitBoxDirection": "normal",
                    "WebkitBoxOrient": "horizontal",
                    "WebkitBoxPack": "center",
                    "WebkitFlexBasis": "auto",
                    "WebkitFlexDirection": "row",
                    "WebkitFlexShrink": 0,
                    "WebkitJustifyContent": "center",
                    "alignItems": "center",
                    "borderBottomStyle": "solid",
                    "borderBottomWidth": "0px",
                    "borderLeftStyle": "solid",
                    "borderLeftWidth": "0px",
                    "borderRightStyle": "solid",
                    "borderRightWidth": "0px",
                    "borderTopStyle": "solid",
                    "borderTopWidth": "0px",
                    "boxSizing": "border-box",
                    "display": "flex",
                    "flexBasis": "auto",
                    "flexDirection": "row",
                    "flexShrink": 0,
                    "justifyContent": "center",
                    "marginBottom": "0px",
                    "marginLeft": "0px",
                    "marginRight": "0px",
                    "marginTop": "0px",
                    "minHeight": "0px",
                    "minWidth": "0px",
                    "msFlexAlign": "center",
                    "msFlexDirection": "row",
                    "msFlexNegative": 0,
                    "msFlexPack": "center",
                    "msFlexPreferredSize": "auto",
                    "paddingBottom": "0px",
                    "paddingLeft": "0px",
                    "paddingRight": "0px",
                    "paddingTop": "0px",
                    "position": "relative",
                    "zIndex": 0,
                  }
                }
              >
                <div
                  dir="auto"
                  style={
                    Object {
                      "borderBottomWidth": "0px",
                      "borderLeftWidth": "0px",
                      "borderRightWidth": "0px",
                      "borderTopWidth": "0px",
                      "boxSizing": "border-box",
                      "color": "rgba(255,255,255,1.00)",
                      "direction": "ltr",
                      "display": "inline",
                      "fontFamily": "Roboto-Medium, Roboto, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
                      "fontSize": "14px",
                      "fontStyle": "inherit",
                      "fontVariant": "inherit",
                      "fontWeight": "inherit",
                      "letterSpacing": "1px",
                      "lineHeight": "inherit",
                      "marginBottom": "9px",
                      "marginLeft": "16px",
                      "marginRight": "16px",
                      "marginTop": "9px",
                      "maxWidth": "100%",
                      "overflowX": "hidden",
                      "overflowY": "hidden",
                      "paddingBottom": "0px",
                      "paddingLeft": "0px",
                      "paddingRight": "0px",
                      "paddingTop": "0px",
                      "textAlign": "center",
                      "textDecoration": "none",
                      "textOverflow": "ellipsis",
                      "whiteSpace": "nowrap",
                      "wordWrap": "break-word",
                    }
                  }
                >
                  <span
                    dir="auto"
                    style={
                      Object {
                        "borderBottomWidth": "0px",
                        "borderLeftWidth": "0px",
                        "borderRightWidth": "0px",
                        "borderTopWidth": "0px",
                        "boxSizing": "border-box",
                        "direction": "ltr",
                        "display": "inline",
                        "fontFamily": "Helvetica, \\"sans-serif\\"",
                        "fontSize": "18px",
                        "fontStyle": "inherit",
                        "fontVariant": "inherit",
                        "fontWeight": "bold",
                        "lineHeight": "inherit",
                        "marginBottom": "0px",
                        "marginLeft": "0px",
                        "marginRight": "0px",
                        "marginTop": "0px",
                        "paddingBottom": "5px",
                        "paddingLeft": "5px",
                        "paddingRight": "5px",
                        "paddingTop": "5px",
                        "textAlign": "center",
                        "textDecoration": "none",
                        "textTransform": "uppercase",
                        "whiteSpace": "inherit",
                        "wordWrap": "break-word",
                      }
                    }
                  >
                    Quick Face Recognition
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
